src/Aspire.Hosting.Azure/Provisioning/Internal/DefaultProvisioningContextProvider.cs

```diff
@@ -26,8 +31,157 @@ internal sealed class DefaultProvisioningContextProvider(
 {
     private readonly AzureProvisionerOptions _options = options.Value;
 
+    private readonly TaskCompletionSource _provisioningOptionsAvailable = new();
```

Comment:
Should this use RunContinuationsAsynchronously? It isn't being set in a lock at the moment, but in case for the futher?